rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -1,163 +1,51 @@
-                    "volumetric_flow_rate", building_total_air_leakage_rate
-                ),
-                "building_total_measured_air_leakage_rate": CalcQ(
-                    "volumetric_flow_rate", building_total_measured_air_leakage_rate
-                ),
-                "target_air_leakage_rate_75pa_p": CalcQ(
-                    "volumetric_flow_rate", target_air_leakage_rate_75pa_p
-                ),
-                "empty_measured_air_leakage_rate_flow_flag": empty_measured_air_leakage_rate_flow_flag,
-            }
-
-        def manual_check_required(self, context, calc_vals=None, data=None):
-            building_total_air_leakage_rate = calc_vals[
-                "building_total_air_leakage_rate"
-            ]
-            target_air_leakage_rate_75pa_p = calc_vals["target_air_leakage_rate_75pa_p"]
-            empty_measured_air_leakage_rate_flow_flag = calc_vals[
-                "empty_measured_air_leakage_rate_flow_flag"
+                if get_opaque_surface_type(surface_b) == OST.BELOW_GRADE_WALL
             ]
+        )
 
-            return (
-                not std_equal(
-                    building_total_air_leakage_rate,
-                    TOTAL_AIR_LEAKAGE_COEFF * target_air_leakage_rate_75pa_p,
-                )
-                and empty_measured_air_leakage_rate_flow_flag
-            )
-
-        def rule_check(self, context, calc_vals=None, data=None):
-            building_total_air_leakage_rate = calc_vals[
-                "building_total_air_leakage_rate"
-            ]
-            building_total_measured_air_leakage_rate = calc_vals[
-                "building_total_measured_air_leakage_rate"
-            ]
-            target_air_leakage_rate_75pa_p = calc_vals["target_air_leakage_rate_75pa_p"]
-            empty_measured_air_leakage_rate_flow_flag = calc_vals[
-                "empty_measured_air_leakage_rate_flow_flag"
-            ]
+    def get_calc_vals(self, context, data=None):
+        rpd = context.BASELINE_0
+        ground_temperature_schedule = rpd["weather"].get("ground_temperature_schedule")
+        return {"ground_temperature_schedule": ground_temperature_schedule}
 
-            return std_equal(
-                building_total_air_leakage_rate,
-                TOTAL_AIR_LEAKAGE_COEFF * target_air_leakage_rate_75pa_p,
-            ) or (
-                building_total_air_leakage_rate
-                != TOTAL_AIR_LEAKAGE_COEFF * target_air_leakage_rate_75pa_p
-                and empty_measured_air_leakage_rate_flow_flag == False
-                and std_equal(
-                    building_total_air_leakage_rate,
-                    TOTAL_AIR_LEAKAGE_COEFF * building_total_measured_air_leakage_rate,
-                )
-            )
+    def applicability_check(self, context, calc_vals, data):
+        ground_temperature_schedule = calc_vals["ground_temperature_schedule"]
+        return ground_temperature_schedule
```

Comment:
I think here you miss the `get_manual_check_required_msg` function to determine the message according to the RDS.
if `ground_temperature_schedule` exist, then return `It cannot be determined if the ground temperature schedule for the project is representative of the project climate.`
else:
`A ground temperature schedule was not found for the project.`

Reply:
Addressed!